File: PruebaPicasso/app/src/main/java/es/iesnervion/avazquez/pruebapicasso/activities/MainActivity.java
package es.iesnervion.avazquez.pruebapicasso.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

import android.os.Bundle;
import android.widget.ImageView;
import android.widget.ListView;

import com.squareup.picasso.Picasso;

import java.util.ArrayList;

import es.iesnervion.avazquez.pruebapicasso.R;
import es.iesnervion.avazquez.pruebapicasso.adapter.ImageAdapter;
import es.iesnervion.avazquez.pruebapicasso.fragments.BreedListFragment;
import es.iesnervion.avazquez.pruebapicasso.fragments.ImageListFragment;
import es.iesnervion.avazquez.pruebapicasso.viewModel.SharedViewModel;

public class MainActivity extends AppCompatActivity {

    Fragment listadoRazas;
    Fragment listadoImagenes;
    SharedViewModel viewModel;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        viewModel = ViewModelProviders.of(this).get(SharedViewModel.class);

        listadoRazas = new BreedListFragment();
        listadoImagenes = new ImageListFragment();

        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();

        ft.replace(R.id.fragment, listadoRazas);
        ft.commit();


        /*El observer*/
        final Observer<String> razaObserver = new Observer<String>() {
            @Override
            public void onChanged(String razaElegida) {
                FragmentTransaction ft2 = getSupportFragmentManager().beginTransaction();
                ft2.replace(R.id.fragment, listadoImagenes).addToBackStack(null).commit();

            }
        };

        //Observo el LiveData con ese observer que acabo de crear
        viewModel.getRazaElegida().observe(this, razaObserver);




    }
}

File: EjemploKotlinRetrofit2/app/src/main/java/es/iesnervion/avazquez/ejemplokotlinretrofit2/activities/MainActivity.kt
package es.iesnervion.avazquez.ejemplokotlinretrofit2.activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import es.iesnervion.avazquez.ejemplokotlinretrofit2.R
import es.iesnervion.avazquez.ejemplokotlinretrofit2.fragments.BreedListFragment
import es.iesnervion.avazquez.ejemplokotlinretrofit2.fragments.ImageListFragment
import es.iesnervion.avazquez.ejemplokotlinretrofit2.viewModel.SharedViewModel


class MainActivity : AppCompatActivity() {

    lateinit var viewModel : SharedViewModel
    lateinit var listaRazasFragment : Fragment
    lateinit var listadoImagenesFragment : Fragment
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
/*
        viewModel = ViewModelProviders.of(this).get(SharedViewModel::class.java)
*/
        /*Fragments*/
  //      listaRazasFragment = BreedListFragment()
        listadoImagenesFragment = ImageListFragment()
        var fm : FragmentManager = supportFragmentManager
        var fragmentTransaction : FragmentTransaction = fm.beginTransaction()

//        fragmentTransaction.replace(R.id.fragment, listaRazasFragment)
        fragmentTransaction.replace(R.id.fragment, listadoImagenesFragment)

        fragmentTransaction.commit()

/*
        // Create the observer which updates the UI.
        val razaElegidaObserver = Observer<String> { observer ->
            // Update the UI
            var fragmentTransaction : FragmentTransaction = fm.beginTransaction()

            fragmentTransaction.replace(R.id.fragment, listadoImagenesFragment).addToBackStack(null)

            fragmentTransaction.commit()
        }

        // Observe the LiveData, passing in this activity as the LifecycleOwner and the observer.
        viewModel.getRazaElegida().observe(this, razaElegidaObserver)


*/




    }


}
